### PR TITLE
Bump WinAppSDK to 1.6

### LINKED ISSUE #11472: [WinUI] First Entry/Editor gets focus if put inside ScrollView
### Description

On WinUI platform, if Entry or Editor are put inside a ScrollView, they do not lose focus correctly on click on empty space or other non-focusable elements. Instead, the first Entry or Editor is focused. 
This same bug was earlier present in Xamarin.Forms (and is still open):
https://github.com/xamarin/Xamarin.Forms/issues/10420

### Steps to Reproduce

1. Create a MAUI App from template 
2. In MainPage inside the VerticalStackLayout put: Entry and/or Editor, some unfocusable element like Label/ContentView and Button. (VerticalStackLayout is already in ScrollView in the template)
3. Focus entry or editor
4. Click on empty space inside ScrollView or on ContentView

### Link to public reproduction project repository

https://github.com/maiia-kuzmishyna/MAUI-EntryScrollViewFocus

### Version with bug

6.0.400

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Windows 11 (SDK 10.0.19041.0)

### Did you find any workaround?

I didn't find any universal workaround, however focus is lost properly on click on Button / other focusable controls.

### Relevant log output

_No response_

### LINKED ISSUE #22894: [Windows] The title bar is shown doubly if I set `Window.AppWindow.TitleBar.ExtendsContentIntoTitleBar` as false.
### Description

In the Windows platform, I observed the window tile bar was shown doubly when I set `Window.AppWindow.TitleBar.ExtendsContentIntoTitleBar` as false. I tried to set this property to false because I wanted to use the native title bar to show a title bar icon with a .ico file. In the current version, it is necessary to switch to the native title bar showing an icon.

### Steps to Reproduce

Set `Window.AppWindow.TitleBar.ExtendsContentIntoTitleBar` as false like the below code:
```cs
builder.ConfigureLifecycleEvents(
	builder => builder.AddWindows(
		builder => builder.OnWindowCreated(window => {
			window.AppWindow.TitleBar.ExtendsContentIntoTitleBar = false;
		})
	)
);
```

**Screenshot for Actual Behavior**
image

**Screenshot for Expected Behavior**
image

### Link to public reproduction project repository

https://github.com/Takym/DoublyTitleBarsInMaui/blob/master/DoublyTitleBarsInMaui/MauiProgram.cs#L31

### Version with bug

8.0.21 SR4.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

Microsoft Windows [Version 10.0.19045.4412]

### Did you find any workaround?

I could hide the second title bar by overlapping it with the native container (`FrameworkElement` object). This code is to set the container margin to `(0, -32, 0, 0)` and the Z index to zero.
```cs
if (window.Content is FrameworkElement elem) {
	elem.Margin = new(0, -32, 0, 0);
	Canvas.SetZIndex(elem, 0);
}
```

I added this code below of `window.AppWindow.TitleBar.ExtendsContentIntoTitleBar = false;` line.
```diff
builder.ConfigureLifecycleEvents(
	builder => builder.AddWindows(
		builder => builder.OnWindowCreated(window => {
			window.AppWindow.TitleBar.ExtendsContentIntoTitleBar = false;

+			if (window.Content is FrameworkElement elem) {
+				elem.Margin = new(0, -32, 0, 0);
+				Canvas.SetZIndex(elem, 0);
+			}
		})
	)
);
```

### Relevant log output

_No response_

### LINKED ISSUE #24875: Bump WinAPPSDK to 1.6

### PR DESCRIPTION
**Pull request by Foda**
### Description of Change

Bump WinAppSDK to 1.6

### Fixes

Fixes #24875 
Fixes #22697 
Fixes #11472 
Fixes #22894

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -12,6 +12,7 @@
     <!--End: Package sources managed by Dependency Flow automation. Do not edit the sources above.-->
     <add key="local" value="LOCAL_PLACEHOLDER" />
     <add key="nuget-only" value="NUGET_ONLY_PLACEHOLDER" />
+    <add key="nuget.org" value="https://api.nuget.org/v3/index.json"/>
```
Comment (Foda) [inline NuGet.config:15]: TODO: remove

File: Directory.Build.props [hunk 2]
```diff
@@ -2,8 +2,8 @@
   <Import Project="Sdk.props" Sdk="Microsoft.DotNet.Arcade.Sdk" />
   <Import Condition="'$(EnvironmentBuildPropsImported)' != 'True'" Project="$(MSBuildThisFileDirectory)eng\Environment.Build.props" />
 
-  <PropertyGroup> 
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  <PropertyGroup>
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (Foda) [inline Directory.Build.props:6]: TODO: remove

File: Directory.Build.targets [hunk 3]
```diff
@@ -81,7 +81,7 @@
 
   <!-- Until we get a new enough dotnet -->
   <ItemGroup Condition="'$(_MauiTargetPlatformIsWindows)' == 'True'">
-    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.19041.31" TargetingPackVersion="10.0.19041.31" />
+    <FrameworkReference Update="Microsoft.Windows.SDK.NET.Ref" RuntimeFrameworkVersion="10.0.19041.35-preview" TargetingPackVersion="10.0.19041.35-preview" />
```
[the remark this answers is not in the document]
Comment (mattleibow) [reply Directory.Build.targets:84]: We need to see what version is shipping with the .NET SDK and use that (or lower) so no-one needs to set this in their project.

[the remark this answers is not in the document]
Comment (mattleibow) [reply Directory.Build.targets:84]: From an insider:

.38 for .NET 6
.41 for .NET 8

[the remark this answers is not in the document]
Comment (mattleibow) [reply Directory.Build.targets:84]: So I think since we support net8 and net9, using 41 will work for both.

[the remark this answers is not in the document]
Comment (mattleibow) [reply Directory.Build.targets:84]: https://github.com/dotnet/sdk/pull/41936

**Comment by Foda** [comment]
@mattleibow Currently blocked on this by: https://github.com/microsoft/WindowsAppSDK/issues/4646

File: Directory.Build.props [hunk 4]
```diff
@@ -123,6 +123,11 @@
     <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
   </PropertyGroup>
 
+  <!-- Windows specific settings -->
+  <PropertyGroup Condition="'$(_MauiTargetPlatformIsWindows)' == 'True'">
+    <WindowsSdkPackageVersion>10.0.19041.41</WindowsSdkPackageVersion>
```
Comment (mattleibow) [inline Directory.Build.props:128]: Should this be 41 or 38?

File: src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj [hunk 5]
```diff
@@ -39,6 +39,10 @@
     <DefineConstants>$(DefineConstants);WINDOWS</DefineConstants>
   </PropertyGroup>
 
+  <ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
+    <PackageReference Include="Microsoft.Web.WebView2" />
```
Comment (mattleibow) [inline src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj:43]: Why is this needed in the sample?

Comment (Foda) [reply src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj:43]: Sample uses WebView2

Comment (Scottj1s) [reply src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj]: > Why is this needed in the sample? I see it is in every project and I am scared...

yes, shouldn't be necessary with WinAppSdk 1.6 now containing a WebView2 nupkg reference.  investigating...

**Comment by Foda** [comment]
Blocked by: https://github.com/microsoft/CsWinRT/issues/1733

File: src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj (hunk 5 above)
Comment (Scottj1s) [reply src/Controls/samples/Controls.Sample/Maui.Controls.Sample.csproj]: circling back - the WebView2 nupkg transitive reference issue will be fixed in the next release of same (later in September)

**Comment by RuddyOne** [comment]
Is this still blocked? We need to PR to potentially fix  #22784

**Comment by symbiogenesis** [comment]
> > You can see release notes for 1.6: https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#version-16
> > There is no reason to wait for this PR to use 1.6.
> 
> And how can one do it please?

Try this:

```csproj
<PropertyGroup>
    <TargetFrameworks>net8.0;net8.0-ios;net8.0-android</TargetFrameworks>
    <TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net8.0-windows10.0.22621.0</TargetFrameworks>
    <WindowsSdkPackageVersion>10.0.22621.38</WindowsSdkPackageVersion>
</PropertyGroup>
```

and

```csproj
  <ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.6.240829007" />
  </ItemGroup>
```

**Comment by RuddyOne** [comment]
> ```
>     <WindowsSdkPackageVersion>10.0.22621.38</WindowsSdkPackageVersion>
> ```

I did not know this, thank you. I will see if I can get this working, getting crashes for Models now. Sure I will get there. Thanks.

File: src/Compatibility/Core/src/Compatibility.csproj [hunk 6]
```diff
@@ -9,7 +9,7 @@
     <iOSRoot>iOS\</iOSRoot>
     <WindowsRoot>Windows\</WindowsRoot>
     <TizenRoot>Tizen\</TizenRoot>
-    <IsAotCompatible Condition="!$(TargetFramework.StartsWith('netstandard'))">true</IsAotCompatible>
+    <IsAotCompatible Condition="!$(TargetFramework.StartsWith('netstandard')) and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'windows'">true</IsAotCompatible>
```
Comment (simonrozsival) [inline src/Compatibility/Core/src/Compatibility.csproj:12]: Why is this change needed?

Comment (Foda) [reply src/Compatibility/Core/src/Compatibility.csproj:12]: Yeah, I didn't want to fix 150+ errors about missing partials then revert it all when RTM was released, so I'm disabling it for now.

Comment (Sergio0694) [reply src/Compatibility/Core/src/Compatibility.csproj:12]: > "then revert it all when RTM was released"

I feel like there might be a misunderstanding here, that requirement is absolutely **not** going away when .NET 9 goes GA. I do have plans to remove that, but that's a very long term thing that needs some C# features that don't even exist yet (extension interfaces, specifically), so that won't happen before .NET 11/C# 15 at the earliest. Until then, the requirement of marking types marshalled to WinRT as partial will remain there 😅

Or did you mean something else by that? Just making sure the long term plan for CsWinRT is clear.

Comment (Foda) [reply src/Compatibility/Core/src/Compatibility.csproj:12]: Ah, OK. I misunderstood that the current analyzers were overly verbose and some combination of .NET 9 + new packages would fix that. 

It's uh... been a bit difficult for us to keep track of what's being fixed/changed with certain packages/SDKs 😋

File: Directory.Build.targets [hunk 7]
```diff
@@ -84,8 +84,11 @@
     <FrameworkReference Include="Microsoft.Windows.SDK.NET.Ref" IsImplicitlyDefined="true" Pack="false" PrivateAssets="All" />
   </ItemGroup>
   <PropertyGroup>
-    <WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>   
+    <WindowsSdkPackageVersion>10.0.19041.34</WindowsSdkPackageVersion>
```
Comment (manodasanW) [inline Directory.Build.targets:87]: Looks like this and above got uncommented during merge.  This is already set to 44 in the props.

Comment (Foda) [reply Directory.Build.targets]: Oops -- lemme remove this

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
- Failing tests are unrelated
